hierarchicalforecast/methods.py

```diff
@@ -324,6 +324,42 @@ def reconcile(
 
     __call__ = reconcile
 
+# Internal Cell
+def optimal_combination(S: np.ndarray,
+                        y_hat: np.ndarray,
+                        method: str,
+                        y_insample: np.ndarray = None,
+                        y_hat_insample: np.ndarray = None):
+
+    return min_trace(S=S, y_hat=y_hat,
+                         y_insample=y_insample,
+                         y_hat_insample=y_hat_insample,
+                         method=method)
+
+# Cell
+class OptimalCombination:
+
+    def __init__(
+            self,
+            method: str # Least Squares Weighting Method
+        ):
+        self.method = method
```

Comment:
It would be nice to have an assert here to consider just the optimal combination methods (`ols` and `wls_struct`).

Reply:
added!